### ISSUE TITLE
False positive for MD053 and MD034

### ISSUE THREAD

**Issue by bronsen** [DavidAnson/markdownlint#2154]
## Problem Description

The following markdown file is rejected by `mdlint check simple.md` but accepted by `markdownlint-cli2 simple.md`.

```markdown
# required title

Here a [reference] is used.

And below it is defined:

[reference]: https://example.com/
```
## mdlint Rejects

`mdlint --version` yields "mdlint 0.3.18"

Here is mdlint rejecting the file:

```text
bronsen@ananke in ~/C/l/teilchensammler-cli on  main via 🐍 (teilchensammler-cli) 
11:37:21 💚❯ mdlint check simple.md
simple.md
  simple.md:7:1: MD053 Link reference definition 'reference' is defined but not used
       | [reference]: https://example.com/
       | ^
  simple.md:7:14: MD034 Bare URL used: https://example.com/
       | [reference]: https://example.com/
       |              ^

Found 2 error(s) in 1 file(s) (1 checked)

```

## markdownlint-cli2 Accepts

`markdownlint-cli2 --help` yields "markdownlint-cli2 v0.23.0 (markdownlint v0.41.0)".

Here is markdownlint-cli2 accepting the file in question:

```text
bronsen@ananke in ~/C/l/teilchensammler-cli on  main via 🐍 (teilchensammler-cli) 
11:37:19 💚❯ markdownlint-cli2 simple.md
markdownlint-cli2 v0.23.0 (markdownlint v0.41.0)
Finding: simple.md
Linting: 1 file(s)
Summary: 0 error(s)

```

## Other Stuff

I believe the file simple.md to be correctly formatted. 

However, in the code base I have seen test files that seem to specifically test against MD053 and MD034: that makes me question the correctness of simple.md.

How do I correctly define and use references (or labels)?

**Comment by DavidAnson** [comment]
I agree that your sample appears correctly formatted.

The CommonMark Dingus agrees as well: https://spec.commonmark.org/dingus/?text=%23%20required%20title%0A%0AHere%20a%20%5Breference%5D%20is%20used.%0A%0AAnd%20below%20it%20is%20defined%3A%0A%0A%5Breference%5D%3A%20https%3A%2F%2Fexample.com%2F%0A

The markdownlint-cli2 tool uses this library (markdownlint) and also agrees, so I think there is no issue for this library/repo.

You may wish to open an issue against mdlint based on that tool's seemingly wrong error report.

**Comment by bronsen** [comment]
oh I was at the wrong repo? sorry about that
